website/blog/2025-06-04-jest-30.md

```diff
@@ -0,0 +1,181 @@
+
+```ts
+import {protectProperties} from 'jest-util';
+
+protectProperties(globalThis['my-property']);
+```
+
+Thanks to [@eyalroth](https://github.com/eyalroth) for implementing this feature!
+
+## New Features
+
+### Improved ECMAScript Module & TypeScript Support
+
+Support for `import.meta.*` and `file://` was added [when using native ESM with Jest](https://jestjs.io/docs/ecmascript-modules). In addition, you can now write your Jest config files in TypeScript, and `.mts` and `.cts` files are natively supported without requiring extra configuration. If you are using Node’s native TypeScript type stripping feature, we no longer load the TypeScript transformer to strip types, leading to faster test runs.
+
+## New & Improved Testing APIs
+
+### Spies and the `using` keyword
+
+You can use JavaScript’s new explicit resource management syntax (`using`) with Jest spies. If your environment supports it, writing `using jest.spyOn(obj, 'method')` will automatically restore the spy when the block ends, so you don’t have to manually clean up. ()
+
+```typescript
+test('logs a warning', () => {
+  using spy = jest.spyOn(console, 'warn');
+  doSomeThingWarnWorthy();
+  expect(spy).toHaveBeenCalled();
+});
+```
+
+[Documentation](/docs/jest-object#spied-methods-and-the-using-keyword)
+
+### `expect.arrayOf`
+
+Jest 30 introduces a new asymmetric matcher, `expect.arrayOf`, which lets you validate every element of an array against a condition or type. For instance, you can expect an array of numbers ensuring all items are numbers:
+
+```javascript
+expect(someArray).toEqual(expect.arrayOf(expect.any(Number)));
+```
+
+[Documentation](/docs/expect#expectarrayofvalue)
+
+### New `test.each` placeholder: `%$`
+
+If you use data-driven tests with `test.each`, you can now include a special placeholder `%$` in your test titles to inject the **number of the test case**. For example:
+
+```javascript
+test.each(cases)('Case %$ works as expected', () => {});
+```
+
+will replace `%$` with the test’s sequence number.
+
+[Documentation](/docs/api#testeachtablename-fn-timeout)
+
+### `jest.advanceTimersToNextFrame()`
+
+[`@sinonjs/fake-timers`](https://github.com/sinonjs/fake-timers) was upgraded to v13, addding `jest.advanceTimersToNextFrame()`. This new function allows you to advance all pending `requestAnimationFrame` callbacks to the next frame boundary, making it easier to test animations or code that relies on `requestAnimationFrame` without having to guess millisecond durations.
+
+[Documentation](/docs/jest-object#jestadvancetimerstonextframe)
+
+### Configurable test retries
+
+Jest 30 enhances `jest.retryTimes()` with new options that give you control on how retries are handled. You can specify a delay or immediately retry a failed test instead of waiting until the entire test suite finishes:
```

Comment:
Maybe it would be useful to add a sentence that describes when these options might be useful, to help contextualize it for folks.